packages/snaps-controllers/src/cronjob/CronjobController.ts

```diff
@@ -152,55 +157,128 @@ export class CronjobController extends BaseController<
+
     // Subscribe to Snap events
-    /* eslint-disable @typescript-eslint/unbound-method */
+    this.#initializeEventListeners();
 
-    this.messagingSystem.subscribe(
-      'SnapController:snapInstalled',
-      this._handleSnapRegisterEvent,
-    );
+    // Register action handlers
+    this.#initializeActionHandlers();
 
-    this.messagingSystem.subscribe(
-      'SnapController:snapUninstalled',
-      this._handleSnapUnregisterEvent,
-    );
+    this.dailyCheckIn().catch((error) => {
+      logError(error);
+    });
 
-    this.messagingSystem.subscribe(
-      'SnapController:snapEnabled',
-      this._handleSnapEnabledEvent,
-    );
+    this.#rescheduleBackgroundEvents(Object.values(this.state.events));
+  }
 
-    this.messagingSystem.subscribe(
-      'SnapController:snapDisabled',
-      this._handleSnapDisabledEvent,
-    );
+  /**
+   * Initialize event listeners.
+   */
+  #initializeEventListeners() {
+    /* eslint-disable @typescript-eslint/unbound-method */
+    const events = {
+      'SnapController:snapInstalled': this._handleSnapRegisterEvent,
+      'SnapController:snapUninstalled': this._handleSnapUnregisterEvent,
+      'SnapController:snapEnabled': this._handleSnapEnabledEvent,
+      'SnapController:snapDisabled': this._handleSnapDisabledEvent,
+      'SnapController:snapUpdated': this._handleEventSnapUpdated,
+    };
+    /* eslint-disable @typescript-eslint/unbound-method */
 
-    this.messagingSystem.subscribe(
-      'SnapController:snapUpdated',
-      this._handleEventSnapUpdated,
-    );
-    /* eslint-enable @typescript-eslint/unbound-method */
+    Object.entries(events).forEach(([event, handler]) => {
+      this.messagingSystem.subscribe(
+        event as CronjobControllerEvents['type'],
+        handler,
+      );
+    });
+  }
 
-    this.messagingSystem.registerActionHandler(
-      `${controllerName}:scheduleBackgroundEvent`,
-      (...args) => this.scheduleBackgroundEvent(...args),
-    );
+  /**
+   * Initialize action handlers.
+   */
+  #initializeActionHandlers(): void {
+    const handlers: Record<BackgroundEventActions, (...args: any[]) => any> = {
+      [`${controllerName}:scheduleBackgroundEvent`]: (
+        event: Omit<BackgroundEvent, 'id' | 'scheduledAt'>,
+      ) => this.scheduleBackgroundEvent(event),
+      [`${controllerName}:cancelBackgroundEvent`]: (
+        origin: string,
+        id: string,
+      ) => this.cancelBackgroundEvent(origin, id),
+      [`${controllerName}:getBackgroundEvents`]: (snapId: SnapId) =>
+        this.getBackgroundEvents(snapId),
+    };
 
-    this.messagingSystem.registerActionHandler(
-      `${controllerName}:cancelBackgroundEvent`,
-      (...args) => this.cancelBackgroundEvent(...args),
-    );
+    Object.entries(handlers).forEach(([action, handler]) => {
+      this.messagingSystem.registerActionHandler(
+        action as BackgroundEventActions,
+        handler,
+      );
+    });
+  }
```

Comment:
I think this is more readable using the previous pattern. That's what we use in other controller's too.

Reply:
Wrapping it in initialise functions is fine, but I'm not convinced by the objects and loops.

Reply:
Changed it back.